tests/filecheck/backend/rvscf_lowering_labels.mlir

```diff
@@ -0,0 +1,112 @@
+// RUN: xdsl-opt -p lower-riscv-scf-to-labels --split-input-file %s | filecheck %s
+
+builtin.module {
+    "riscv_func.func"() ({
+    ^0(%0 : !riscv.reg<a0>, %1 : !riscv.reg<a1>):
+        %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<a0>
+        %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a1>
+        %4 = riscv.li 1 : () -> !riscv.reg<a2>
+        %5 = riscv.li 0 : () -> !riscv.reg<a3>
+        %6 = riscv.fcvt.s.w %5 : (!riscv.reg<a3>) -> !riscv.freg<fa0>
+        %7 = "riscv_scf.for"(%2, %3, %4, %6) ({
+        ^1(%8 : !riscv.reg<a0>, %9 : !riscv.freg<fa0>):
+            %10 = riscv.fcvt.s.w %8 : (!riscv.reg<a0>) -> !riscv.freg<fa1>
+            %11 = riscv.fadd.s %9, %10 : (!riscv.freg<fa0>, !riscv.freg<fa1>) -> !riscv.freg<fa0>
+            "riscv_scf.yield"(%11) : (!riscv.freg<fa0>) -> ()
+        }) : (!riscv.reg<a0>, !riscv.reg<a1>, !riscv.reg<a2>, !riscv.freg<fa0>) -> !riscv.freg<fa0>
+        %12 = riscv.fcvt.w.s %7 : (!riscv.freg<fa0>) -> !riscv.reg<a0>
+        "riscv_func.return"(%12) : (!riscv.reg<a0>) -> ()
+    }) {"sym_name" = "foo"} : () -> ()
+}
+
+// CHECK:       builtin.module {
+// CHECK-NEXT:    "riscv_func.func"() ({
+// CHECK-NEXT:    ^0(%0 : !riscv.reg<a0>, %1 : !riscv.reg<a1>):
+// CHECK-NEXT:      %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<a0>
+// CHECK-NEXT:      %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a1>
+// CHECK-NEXT:      %4 = riscv.li 1 : () -> !riscv.reg<a2>
+// CHECK-NEXT:      %5 = riscv.li 0 : () -> !riscv.reg<a3>
+// CHECK-NEXT:      %6 = riscv.fcvt.s.w %5 : (!riscv.reg<a3>) -> !riscv.freg<fa0>
+// CHECK-NEXT:      %7 = riscv.get_register : () -> !riscv.reg<a0>
+// CHECK-NEXT:      riscv.label "scf_cond_0_for" ({
+// CHECK-NEXT:      }) : () -> ()
+// CHECK-NEXT:      riscv.bge %7, %3, "scf_body_end_0_for" : (!riscv.reg<a0>, !riscv.reg<a1>) -> ()
```

Comment:
I guess the riscv.get_register() is how we move SSA-values around branches. That's an interesting approach that is different to what we did in ChocoPy. It mixes SSA-semantics within BBs and side-effectful semantics. I guess it also means that all instructions that write to register-allocated types are suddenly side-effectful as they will have write effects. Has this been documented somewhere? I feel getting this consistent might be difficult.